modules/todo.js

```diff
@@ -0,0 +1,97 @@
+import { parseInt } from 'lodash';
+import Selector from './selectors.js';
+import checkStatus from './statusCheck.js';
+
+const ObjectSelec = new Selector();
+export default class Todo {
+  constructor() {
+    this.todos = JSON.parse(localStorage.getItem('todos')) || [];
+  }
+
+  visable = () => {
+    if (this.todos.length === 0) {
+      ObjectSelec.clear.classList.add('display');
+    } else {
+      ObjectSelec.clear.classList.remove('display');
+    }
+  }
+
+  dispaly = () => {
+    ObjectSelec.todoBody.querySelector('.js-todo-list').innerHTML = '';
+    this.todos.forEach((todo) => {
+      const todoList = `<li data-id="${todo.id}" data-status="${todo.status}">
+      <label for="${todo.id}">
+      <input type="checkbox" class="check" id="${todo.id}" value="${todo.id}"
+      ${todo.status === 'completed'}/>
+      <input type="text" class="js-edit-input" value="${todo.task}"/>
+      </label>
+          <div class="action">
+          <button class="js-delete">
+              <i class="ri-delete-bin-fill"></i>
+          </button>
+          </div> 
+      </li>`;
+      ObjectSelec.todoBody.querySelector('.js-todo-list').innerHTML += todoList;
+    });
+    this.visable();
+  }
+
+  addTask = (e) => {
+    e.preventDefault();
+    const todo = ObjectSelec.input.value;
+    if (!todo) return;
+    if (this.todos.length === 0) {
+      const myTodo = { id: 1, task: todo, status: false };
+      this.todos.push(myTodo);
+    } else {
+      const index = this.todos.length + 1;
+      const myTodo = { id: index, task: todo, status: false };
+      this.todos.push(myTodo);
+    }
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+    ObjectSelec.input.value = '';
+    this.dispaly();
+  }
+
+  deletTask = async (e) => {
+    const btndelete = e.target.closest('.js-delete');
```

Comment:
As you did good design on selectors, you created a class for selectors [selectors.js](https://github.com/AliShahSafdari/Todo-List-Best-Practice-/pull/1/commits/9778355aa38e069d0162eacd64e22d63bbcb150b#diff-f75e8e263e0fd6b49b87852628681f4a98a04200c705d1e8daa7000b2927bb4d). this dellete button selector should be on selectos class.

Reply:
Thanks